widgetssdk/src/main/java/com/glia/widgets/engagement/completion/EngagementCompletionActivityWatcher.kt

```diff
@@ -76,7 +76,10 @@ internal class EngagementCompletionActivityWatcher(
         if (actionOnEnd == Engagement.ActionOnEnd.UNKNOWN) {
             Logger.w(TAG, "Engagement ended with unknown case.")
         }
-        if (isEndedByVisitor || actionOnEnd == Engagement.ActionOnEnd.RETAIN || actionOnEnd == Engagement.ActionOnEnd.SHOW_SURVEY) {
+        if (isEndedByVisitor) {
```

Comment:
Could you please replace this `if-else` chain with `when`

Reply:
Is there a difference in readability or performance?

Reply:
I think `when` is more readable.

Reply:
Converted to `when` but I think the way I did it it is hard to say for sure, at least for me, if readability improved. Could you check this change and see if there are any improvements that I missed? Otherwise, right now I think it is a question of preference.

Reply:
Looks better to me. Thanks 👍.